File: billing_ui_app/src/app/core/models/billing/revspring-payload.response.ts
export interface RevSpringPayloadResponse {
  payload: {
    consumerNumber: string;
    externalUsername: string | null;
    userEmail: string | null;
    userLastName: string | null;
    roleName: string | null;
    accessLevel: string | null;
    accountId: number;
    memberId: number;
    referenceNo : string | null;
    OrgSiteName:string | null;
    dataContext: {
      consumer: {
        consumerNumber: string;
        firstName: string | null;
        lastName: string | null;
        address1: string | null;
        address2: string | null;
        country: string | null;
        city: string | null;
        state: string | null;
        zip: string | null;
        phone: string | null;
        email: string | null;
        dateOfBirth: string | null;
        amountDue: number | null;
      };
    };
  };
}

File: Microservices/BillingService/BillingService.Domain/Models/PaymentPosting/RevSpringPayloadResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BillingService.Domain.Models.PaymentPosting
{
    public class RevSpringPayloadResponse
    {
        public RevSpringPayload Payload { get; set; }
    }
    public class RevSpringPayload
    {
        public string ConsumerNumber { get; set; }
        public string ExternalUsername { get; set; }
        public string UserEmail { get; set; }
        public string UserLastName { get; set; }
        public string RoleName { get; set; }
        public string AccessLevel { get; set; }
        public string PatientFirstName { get; set; } = string.Empty;
        public string PatientLastName { get; set; } = string.Empty;
        public string OrgSiteName { get; set; }
        public string PatientId { get; set; }

        public int AccountId { get; set; }
        public int MemberId { get; set; }
        public string ReferenceNo { get; set; }
        public RevSpringDataContext DataContext { get; set; }
    }
    public class RevSpringConsumer
    {
        public string ConsumerNumber { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }

        public string Address1 { get; set; } = string.Empty;
        public string Address2 { get; set; } = string.Empty;
        public string Country { get; set; }
        public string City { get; set; }
        public string State { get; set; }
        public string Zip { get; set; }

        public string Phone { get; set; }
        public string Email { get; set; }

        public string? DateOfBirth { get; set; }

        public decimal AmountDue { get; set; }
    }

    public class RevSpringDataContext
    {
        public RevSpringConsumer Consumer { get; set; }
    }




}
